Line-wise transforms (#699)
sindresorhus/execa: docs/transform.md

```js
import {execa} from 'execa';

const transform = async function * (lines) {
	for await (const line of lines) {
		const prefix = line.includes('error') ? 'ERROR' : 'INFO'
		yield `${prefix}: ${line}`
	}
};

const {stdout} = await execa('echo', ['hello'], {stdout: transform});
console.log(stdout); // HELLO
```

> Alternatively, a `{ transform }` plain object can also be passed.

Reviewer: I don't see the point in documenting this. It's only there for the `binary` option and that option is already documented further down.

Author: Yes, I was hesitating too whether to include this.
Fixed in 51881563dec1a20ad9d336d0c20c49908c887121

console.log(stdout); // ''
```

## Binary data

The transform iterates over lines by default.\
However, if the `binary` transform option is `true`, it iterates over arbitrary chunks of data instead.

```js
await execa('./binary.js', {stdout: {transform, binary: true}});
```

This is more efficient and recommended if the data is either:

>   - Binary, i.e. it does not have lines.
> 	- Text, but the transform works even if a line or word is split across multiple chunks.

Reviewer: ```suggestion
	- Binary: Which does not have lines.
	- Text: But the transform works even if a line or word is split across multiple chunks.
```
